### PR TITLE
Added function to remove a form group as suggested in Issue #2294

### PR DESCRIPTION
**Pull request by core23**
Rebased https://github.com/sonata-project/SonataAdminBundle/pull/2996/

### CODE REVIEW

**Comment by core23** [comment]
Ping @OskarStark

File: Resources/doc/cookbook/recipe_delete_field_group.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                $formMapper->remove('twitterName');
+                $formMapper->remove('gplusUid');
+                $formMapper->remove('gplusName');
+
+            }
+        }
+
+
+This works, as long as the extended Admin does not use Groups to organize it's field. In the above example, we try to remove all fields from the User Admin, that comes with the SonataUserBundle.
+However, since the fields we deleted, are all part of the 'Social' Group of the form, the fields will be deleted, but the group will stay, even though it is empty.
```
Comment (OskarStark) [inline Resources/doc/cookbook/recipe_delete_field_group.rst]: ``..deleted and the empty group will stay.``

File: Form/FormMapper.php [hunk 2]
```diff
@@ -174,6 +174,46 @@ public function remove($key)
 
     /**
      * @return FormBuilderInterface
+     *                              Removes a group.
+     *
+     * @param string $group              The group to delete
+     * @param string $tab                The tab the group belongs to, defaults to 'default'
+     * @param bool   $deleteTabWhenEmpty Wether or not the Tab should be deleted, when the deleted group leaves the tab emmpty after deletion
+     */
+    public function removeGroup($group, $tab = 'default', $deleteTabWhenEmpty = false)
```
Comment (OskarStark) [inline Form/FormMapper.php]: ``$deleteEmptyTab`` is better IMO

File: Resources/doc/cookbook/recipe_delete_field_group.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+For this case, the FormMapper comes with a method, which allows you to get rid of a whole form group: ``removeGroup``
```
Comment (OskarStark) [inline Resources/doc/cookbook/recipe_delete_field_group.rst]: FormMapper in double ticks

**Comment by OskarStark** [comment]
After the comments this PR is RTM

**Comment by core23** [comment]
RTM @OskarStark

**Comment by OskarStark** [comment]
Awesome @core23

**Pull request merged**